pkg/xray/resource_xray_watch.go

```diff
@@ -0,0 +1,117 @@
+package xray
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceXrayWatch() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceXrayWatchCreate,
+		ReadContext:   resourceXrayWatchRead,
+		UpdateContext: resourceXrayWatchUpdate,
+		DeleteContext: resourceXrayWatchDelete,
+		Description:   "Provides an Xray watch resource.",
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the watch (must be unique)",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Description of the watch",
+			},
+			"active": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Whether or not the watch will be active",
+			},
+			"watch_resource": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Description: "Nested argument describing the resources to be watched. Defined below.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:             schema.TypeString,
+							Required:         true,
+							Description:      "Type of resource to be watched. Options: `all-repos`, `repository`, `build`, `project`, `all-projects`.",
+							ValidateDiagFunc: inList("all-repos", "repository", "build", "project", "all-projects"),
+						},
+						"bin_mgr_id": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Default:     "default",
+							Description: "The ID number of a binary manager resource. Should be set to `default` if not set on the Artifactory side.",
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The name of the build or repository. Enable Xray indexing must be enabled on the repo or build",
+						},
+						"filter": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Description: "Nested argument describing filters to be applied. Defined below.",
```

Comment:
Can you put a validator here? You want to validate the value for their types

Reply:
I have validation for the type `ValidateDiagFunc: inList("regex", "package-type", "ant-patterns"),`